internal/pkg/manifest/validate.go

```diff
@@ -1455,7 +1457,20 @@ func (q SQSQueueOrBool) validate() error {
 	if q.IsEmpty() {
 		return nil
 	}
-	return q.Advanced.validate()
+
+	if q.Enabled != nil {
+		return nil
+	} else {
+		if q.Advanced.Type != nil && strings.Compare(aws.StringValue(q.Advanced.Type), "fifo") == 0 {
+			return q.Advanced.validateFIFO()
+		} else {
+			if q.Advanced.FifoThroughputLimit != nil || q.Advanced.HighThroughputFifo != nil ||
```

Comment:
Consider decomposing complex conditionals to a separate method to make the body easier to parse: 
```go
func (q SQSQueue) hasFIFOFieldsEnabled() bool {
   return q.Advanced.FifoThroughputLimit != nil || q.Advanced.HighThroughputFifo != nil // || ...
}
```


https://refactoring.com/catalog/decomposeConditional.html